src/entities/permit2Helper/index.ts

```diff
@@ -323,15 +323,16 @@ const signPermit2 = async (
 const getDetails = async (
     client: Client & PublicActions,
     token: Address,
-    owner: Address,
+    owner: Address | Account,
     spender: Address,
     amount = MaxAllowanceTransferAmount,
     expiration = Number(MaxAllowanceExpiration),
     nonce?: number,
 ) => {
+    const _owner = typeof owner === 'string' ? owner : owner.address;
     let _nonce: number;
     if (nonce === undefined) {
-        _nonce = await getNonce(client, token, owner, spender);
+        _nonce = await getNonce(client, token, _owner, spender);
```

Comment:
@brunoguerios lmk if you agree with this approach and if we should handle the rest of the sign helper methods as part of the PR or open a new issue

Reply:
Sure! It does seem a more robust approach. If it's just a matter of adding that single line for `owner` type check on each method, I'd go ahead and make the change 👍